### ISSUE TITLE
I have modified your model, Will there be any change now?

### ISSUE THREAD

**Issue by AnuragIISC** [xiaomi1024/code_SAMS#5]
```
class SAMS(nn.Module):
    def __init__(self, num_classes, in1, in2):
        super(SAMS, self).__init__()
        #####
        self.hidden_size1 = 128
        
        self.attention_size = 4
        self.num_layers = 2
        self.direction = 2
        self.gru1 = nn.GRU(in1, self.hidden_size1, self.num_layers, batch_first=True, bidirectional=True)        
        self.gru2 = nn.GRU(in2, self.hidden_size1, self.num_layers, batch_first=True, bidirectional=True)
        

        self.Att1 = Att(self.hidden_size1, self.attention_size, self.num_layers)
        



        self.st_all_c = nn.Sequential(
            nn.Linear((self.direction * self.hidden_size1 + self.direction * self.hidden_size1) * 2, 64),
            nn.ReLU())
        self.st = nn.Sequential(
            nn.Linear((self.direction * self.hidden_size1 + self.direction * self.hidden_size1) *2, 64),
            nn.ReLU())
        self.s = nn.Sequential(
            nn.Linear(self.direction * self.hidden_size1, 64),
            nn.ReLU())
        self.t = nn.Sequential(
            nn.Linear(self.direction * self.hidden_size1, 64),
            nn.ReLU())
        self.ss = nn.Sequential(
            nn.Linear(self.direction * self.hidden_size1, 64),
            nn.ReLU(),
            nn.Linear(64, num_classes))
        self.tt = nn.Sequential(
            nn.Linear(self.direction * self.hidden_size1, 64),
            nn.ReLU(),
            nn.Linear(64, num_classes)
        )
        self.c_fc = nn.Linear(64, num_classes)

    def mul(self, a, b):
        attn = torch.bmm(a, b.permute(0, 2, 1))
        output = torch.bmm(F.softmax(attn, 2), b)
        return output
    def mut(self, q, k):
        attn = torch.bmm(q.unsqueeze(2), k.unsqueeze(1))
        output = torch.bmm(F.softmax(attn, 2),  k.unsqueeze(2)).view(q.shape[0],-1) + q
        return output
    def forward(self, x, xf, training, stf):
        x = x[ : , :, 0 : 78]
        
        if stf == 'st':
            # # #################### GRU ##############################################
            h1 = torch.zeros(self.num_layers * self.direction, x.size(0), self.hidden_size1).to(device)  # 2 for bidirection
            
            #################feature learning########################
            out1, _ = self.gru1(x, h1)
            
            out2, _ = self.gru2(xf, h1)
            
           
            
            

            ################IFM1#################
            out_s = self.mul(out1, out1) + out1
           
            out_t = self.mul(out2, out2) + out2 
            
            
           


            outa1 = self.Att1(x, out1)
            
            outa2 = self.Att1(xf, out2)
            
            
            

            out_s1 = self.Att1(x, out_s)
            out_t1 = self.Att1(xf, out_t)
            
            

            ################IFM2#################
            a = self.mut(outa1, outa1)
            b = self.mut(outa2, outa2)
            
            

            c1 = torch.cat((a, b), 1)
           
            
            c2 = torch.cat((out_s1, out_t1), 1)
            


            # #######ALL center #####
            #c_all_c = torch.cat((torch.cat((c1, c2), 1), torch.cat((c1, c2), 1)), 1)
            c_all_c = (torch.cat((c1, c2), 1))
            
            out = self.st_all_c(c_all_c)

            out = self.c_fc(out)
            if training == 1:
                return out, outa1, outa2, outa1, outa2, self.ss(outa1), self.tt(outa2)
            else:
                return out
```

Why are you using same parameter again and again by different names?

**Comment by Coding511** [comment]
Agreed the dimensions of all redundant terms are same, and also different learning in one epoch. It should work same
Are you getting same result 78.1 % weighted accuracy on IEMOCAP?
